File: uil_frontend_react/src/lib/enums.ts
/**
 * User
 */
export enum UserRole {
  SUPER_ADMIN = "Super Admin",
  ADMIN = "Admin",
  STAFF = "Staff",
  INDUSTRY = "Industry",
}

export enum UserStatus {
  PENDING = "PENDING",
  APPROVED = "APPROVED",
  REJECTED = "REJECTED",
}

/**
 * Industry
 */
export enum IndustryType {
  IT = "it",
  Manufacturing = "manufacturing",
  Construction = "construction",
  Healthcare = "healthcare",
  Education = "education",
  Other = "other",
}

/**
 * Request
 */
export enum IndustryRequestType {
  RND = "rnd",
  TECH_SUPPORT = "tech_support",
  CONSULTANCY = "consultancy",
  TESTING = "testing",
  TRAINING = "training",
  INTERNSHIP = "internship",
  EXTERNSHIP = "externship",
  RECRUITMENT = "recruitment",
  OTHER = "other",
}

export enum OfficeRequestType {
  CURRICULUM_REVIEW = "curriculum_review",
  INDUSTRIAL_VISIT = "industrial_visit",
  JOINT_RESEARCH = "joint_research",
  GUEST_LECTURE = "guest_lecture",
  LAB_ACCESS = "lab_access",
  TECH_TRANSFER = "tech_transfer",
  WORKSHOP_CALL = "workshop_call",
  CONFERENCE_CALL = "conference_call",
  EXHIBITION_CALL = "exhibition_call",
  JOINT_COMMUNITY_ENGAGEMENT = "joint_community_engagement",
  OTHER = "other",
}

export enum Entity {
  INDUSTRY = "industry",
  ACADEMIC_UNIT = "academic_unit",
  STAFF = "staff",
  STUDENT = "student",
}

export enum ActionType {
  INITIATED = "initiated",
  ASSIGNED = "assigned",
  FORWARDED = "forwarded",
  ACCEPT_FORWARDED = "accept_forwarded",
  POSTED_AS_THEMATIC = "posted_as_thematic",
  REPLIED = "replied",
  REJECTED = "rejected",
  // REASSIGNED = "reassigned",
  COMPLETED = "completed",
  // REVOKED = "revoked",
  CANCELLED = "cancelled",
  REVERTED = "reverted",
}

/**
 * Org Units
 */
export enum OrgUnitType {
  UNIVERSITY = "university",
  CAMPUS = "campus",
  COLLEGE = "college",
  INSTITUTE = "institute",
  FACULTY = "faculty",
  SCHOOL = "school",
  DEPARTMENT = "department",
  ACADEMY = "academy",
  RESEARCH_CENTER = "research_center",
  SUPPORT_UNIT = "support_unit",
  PROGRAM = "program",
  OFFICE = "office",
  OTHER = "other",
}

/**
 * Assignments
 */
export enum AssignmentStatus {
  PENDING = "pending",
  ACCEPTED = "accepted",
  REJECTED = "rejected",
  IN_PROGRESS = "in_progress",
  COMPLETED = "completed",
  CANCELLED = "cancelled",
}

/**
 * Posts
 */
export enum PostType {
  SUCCESS_STORY = "success_story",
  THEMATIC_AREA = "thematic_area",
  OPEN_REQUEST = "open_request",
  ANNOUNCEMENT = "announcement",
  GUIDELINE = "guideline",
}

export enum PostContentType {
  REQUEST = "request",
  ASSIGNMENT = "assignment",
  USER = "user",
  INDUSTRY_PROFILE = "industryprofile",
}

File: uil_backend/industry_linkage/enums.py
from django.db import models


class RequestingEntity(models.TextChoices):
    INDUSTRY = "industry", "Industry"
    ACADEMIC_UNIT = "academic_unit", "Academic Unit"
    STAFF = "staff", "Staff"
    STUDENT = "student", "Student"


class RequestType(models.TextChoices):
    # industry requests
    RND = "rnd", "Research & Development Services"
    TECH_SUPPORT = "tech_support", "Technology Support"
    CONSULTANCY = "consultancy", "Consultancy"
    TESTING = "testing", "Testing & QA"
    TRAINING = "training", "Training"
    INTERNSHIP = "internship", "Internship/Externship"
    RECRUITMENT = "recruitment", "Graduate Recruitment"

    # university requests
    CURRICULUM_REVIEW = "curriculum_review", "Curriculum Review"
    INDUSTRIAL_VISIT = "industrial_visit", "Industrial Visit"
    JOINT_RESEARCH = "joint_research", "Joint Research"
    GUEST_LECTURE = "guest_lecture", "Guest Lecturing"
    LAB_ACCESS = "lab_access", "Equipment / Lab Access"
    TECH_TRANSFER = "tech_transfer", "IP / Technology Transfer"
    WORKSHOP_CALL = "workshop_call", "Workshop Call"
    CONFERENCE_CALL = "conference_call", "Conference Call"
    EXHIBITION_CALL = "exhibition_call", "Exhibition Call"
    JOINT_COMMUNITY_ENGAGEMENT = "joint_community_engagement", "Joint Community Engagement"
    OTHER = "other", "Other"


class ActionTypes(models.TextChoices):
    INITIATED = "initiated", "Initiated"
    ASSIGNED = "assigned", "Assigned"
    FORWARDED = "forwarded", "Forwarded"
    POSTED_AS_THEMATIC = "posted_as_thematic", "Posted as Thematic Call"
    REJECTED = "rejected", "Rejected"
    COMPLETED = "completed", "Completed"
    CANCELLED = "cancelled", "Cancelled"
    ACCEPT_FORWARDED = "accept_forwarded", "Accept Forwarded"
    REPLIED = "replied", "Replied"
    REVERTED = "reverted", "Reverted"


class AssignmentStatus(models.TextChoices):
    PENDING = "pending", "Pending"
    ACCEPTED = "accepted", "Accepted"
    REJECTED = "rejected", "Rejected"
    IN_PROGRESS = "in_progress", "In Progress"
    COMPLETED = "completed", "Completed"
    CANCELLED = "cancelled", "Cancelled"


ENTITY_MAP = {
    RequestingEntity.STAFF: {
        "app": "auth",
        "model": "user",
    },
    RequestingEntity.STUDENT: {
        "app": "auth",
        "model": "user",
    },
    RequestingEntity.INDUSTRY: {
        "app": "industry_linkage",
        "model": "industry",
    },
    RequestingEntity.ACADEMIC_UNIT: {
        "app": "organizational_structure",
        "model": "organizationalunit",
    },
}


ACTION_TRANSITIONS = {
    ActionTypes.INITIATED: [],

    ActionTypes.FORWARDED: [
        ActionTypes.ACCEPT_FORWARDED,
        ActionTypes.REJECTED,
    ],

    ActionTypes.ACCEPT_FORWARDED: [
        ActionTypes.REVERTED,
    ],

    ActionTypes.ASSIGNED: [
        ActionTypes.REVERTED,
    ],


    ActionTypes.POSTED_AS_THEMATIC: [
        ActionTypes.REVERTED,
    ],


    ActionTypes.REJECTED: [
        ActionTypes.REVERTED,
    ],

    ActionTypes.COMPLETED: [
        ActionTypes.REVERTED,
    ],

    ActionTypes.CANCELLED: [
        ActionTypes.REVERTED,
    ],

    ActionTypes.REPLIED: [
        ActionTypes.REPLIED,
        ActionTypes.REVERTED,
    ],

    ActionTypes.REVERTED: [],  
}



ACTION_TRANSITIONS = {
    ActionTypes.INITIATED: [],

    ActionTypes.FORWARDED: [
        ActionTypes.ACCEPT_FORWARDED,
        ActionTypes.REJECTED,
    ],

    ActionTypes.ACCEPT_FORWARDED: [
        ActionTypes.REVERTED,
    ],

    ActionTypes.ASSIGNED: [
        ActionTypes.REVERTED,
    ],



    ActionTypes.POSTED_AS_THEMATIC: [
        ActionTypes.REVERTED,
    ],


    ActionTypes.REJECTED: [
        ActionTypes.REVERTED,
    ],

    ActionTypes.COMPLETED: [
        ActionTypes.REVERTED,
    ],

    ActionTypes.CANCELLED: [
        ActionTypes.REVERTED,
    ],

    ActionTypes.REPLIED: [
        ActionTypes.REPLIED,
        ActionTypes.REVERTED,
    ],

    ActionTypes.REVERTED: [],  
}